[ADD] l10n_gt: add Guatemalan fiscal localization (#14190)
odoo/documentation: content/applications/finance/fiscal_localizations/guatemala.rst

=========
Guatemala
=========

.. |SAT| replace:: :abbr:`SAT (Superintendencia de Administración Tributaria)`
.. |EDI| replace:: :abbr:`EDI (Electronic Data Interchange)`

.. _guatemala/intro:

Introduction
============

With the Guatemalan localization, you can generate electronic documents with its XML, fiscal folio,
electronic signature and connection to tax authority Superintendencia de Administración Tributaria
(SAT) through Infile.

The supported documents are:

- :guilabel:`FACT-Factura`, :guilabel:`FCAM-Factura Cambiaria`, :guilabel:`NCRE-Credit Note`,
  :guilabel:`NDEB-Debit Note`;
- :guilabel:`NABN-Nota de Abono`, :guilabel:`FPEQ-Factura de Pequeño Contribuyente`,
  :guilabel:`FCAP-Factura Cambiaria Pequeño Contribuyente`;

> - :guilabel:`FACT-Factura with Export Complement`.

Reviewer: Updated list and punctuation now @auva-odoo

- :guilabel:`VAT Affiliation`: Select the VAT affiliation for the company, which is the type of
  Regime the company belongs to.
- :guilabel:`Legal Name`: The legal name of the company, which is used in the XML and PDF documents.
- :guilabel:`Establishment Code`: A necessary part of the XML when creating an electronic document.
  If this field is not set, all electronic documents will be rejected.

  To find the :guilabel:`Establishment Code`, follow these steps:

  #. From your `SAT account <https://portal.sat.gob.gt/>`_, go to :menuselection:`FEL -->
     Administración de Establecimientos`.
  #. You will see the list of establishments registered along with their corresponding code.

.. important::
   After configuring the company in the database settings, navigate to :menuselection:`Contacts` and
   verify that the :guilabel:`Identification Number` :guilabel:`Type` is set to :guilabel:`NIT`.

In the :guilabel:`Sales & Purchase` tab, select the appropriate phrase in the :guilabel:`Phrases`
field under :guilabel:`Fiscal Information`; this phrase defines the VAT affiliation in the XML and
is also used in the generated PDF.

> Electronic invoice data

Reviewer: ```suggestion
Electronic Invoice credentials
```
It will facilitate keywords/search for basic setup

> -----------------------
> 
> - Sign a service agreement directly with `Infile <https://infile.com.gt/>`_.

Reviewer 2: ```suggestion
-----------------------

In Guatemala, electronic invoicing is mandatory for most businesses. Odoo connects with the authorized provider Infile to generate and submit electronic documents to the SAT for validation. Before issuing electronic documents, you must configure and connect Odoo to Infile to ensure they are properly validated and assigned an official UUID.

**Infile process**
- Contact and sign a service agreement directly with `Infile <https://infile.com.gt/>`_.
```

- Infile will provide you with the necessary credentials to input in Odoo.

> - In Odoo, navigate to :menuselection:`Accounting --> Configuration --> Settings` and scroll down to
>   the :guilabel:`Guatemalan Localization` section.
> - Select the :guilabel:`Infile Web Services` environment, either :guilabel:`Production` or
>   :guilabel:`Testing`.
> - Enter the :guilabel:`Infile Data`:
>   - :guilabel:`Infile WS Password`
>   - :guilabel:`Commerce Code`
>   - :guilabel:`Terminal Code`
> - Click on :guilabel:`Save`.

Reviewer: <img width="1037" height="763" alt="image" src="https://github.com/user-attachments/assets/d39d4a0e-929d-4848-b910-9a485e4d00a6" />
Reference for clarification in other localizations so the user knows what process are required outside Odoo

```suggestion
In Odoo, once you have completed the Infile process, navigate to :menuselection:`Accounting --> Configuration --> Settings` and scroll down to
  the :guilabel:`Guatemalan Localization` section.
  
- Select the :guilabel:`Infile Web Services` environment, either :guilabel:`Production` or
  :guilabel:`Testing`.
- Enter the :guilabel:`Infile Data`:
  - :guilabel:`Infile WS Password`
  - :guilabel:`Commerce Code`
  - :guilabel:`Terminal Code`
- Click on :guilabel:`Save`.
```

> 

Reviewer 3: We follow [sentence case in headings](https://github.com/odoo/documentation/pull/14194) :wink: 
As for the image, personally, I don’t think it’s necessary at all. If someone wants to locate the setting, they can simply type its name in the search bar. That should be straightforward enough.

> 

Author: Instructions unclear and missing consensus, resolving comment

enable the :guilabel:`Send to SAT` and :guilabel:`by Email` checkboxes to send the XML to the |SAT|
through Infile's web service and the validated invoice to the client email and click
:guilabel:`Send`. Then, the following occurs:

- The XML document is created.
- The :abbr:`UUID (Universally Unique Identifier)` is generated.
- The XML is processed synchronously by Infile.

  - If accepted, the file is displayed in the chatter and the email to the client with the
    corresponding :file:`pdf` and :file:`xml` file is sent.
  - If the file contains errors, a warning message displays the reason(s) and the email is not sent.

.. image:: guatemala/pdf-xml-chatter-guatemala.png
   :alt: EDI documents available in the chatter.

The :guilabel:`SAT` tab then displays the following:

- :guilabel:`Datetime`: Timestamp recorded of the XML creation.
- :guilabel:`GT Status`: Status result obtained in the |SAT| response. If the file contains errors,
  a warning message displays the reason(s) and the email is not sent.
- :guilabel:`UUID`: The unique identifier assigned by the |SAT| to the electronic document.

> - :guilabel:`Download`: To download the sent XML file, even if the |SAT| result was rejected.

Reviewer 2: ```suggestion
- :guilabel:`Download Certificate`: To download the sent XML file, even if the |SAT| result was rejected.
```

.. image:: guatemala/sat-tab-electronic-document.png
   :alt: EDI document record available in SAT tab.

.. _localization/guatemala/credit-notes:

Customer electronic notes
-------------------------

> 

Author: Should be fixed with this change request: https://github.com/odoo/documentation/pull/14190#discussion_r2252514364. Resolving...

> Customer electronic notes
> -------------------------
> 
> The :doc:`Customer credit note <../accounting/customer_invoices/credit_notes>` is an electronic
> document that, when validated, is sent to |SAT| via Infile. It is necessary to have a validated
> (posted) electronic invoice to register a credit note. On the invoice, click the :guilabel:`Credit
> note` button to access the :guilabel:`Create credit note` form, then complete the following
> information:
> 
> - :guilabel:`Reason`: Type the reason for the credit note.
> - :guilabel:`Journal`: Select the journal.
> - :guilabel:`Reversal Date`: Type the date.
> 
> The :doc:`Customer debit note <../accounting/customer_invoices/credit_notes>` is an electronic
> document that, when validated, is sent to |SAT| via Infile. It is necessary to have a validated
> (posted) electronic invoice to register a debit note. On the invoice, click the :icon:`fa-cog`
> (:guilabel:`action menu`) icon, select the :guilabel:`Debit note` option to access the
> :guilabel:`Create credit note` form, then complete the following information:
> 
> - :guilabel:`Reason`: Type the reason for the debit note.
> - :guilabel:`Journal`: Select the journal.
> - :guilabel:`Copy lines`: Tick the checkbox to copy the invoice lines to the debit note.
> - :guilabel:`Debit note date`: Type the date.
> 
> .. note::
>    Confirm the invoice to create it. To send the document to |SAT| via Infile, click on
>    :guilabel:`Send` and select the checkbox :guilabel:`Send to SAT`.

Reviewer 2: ```suggestion
Debit and credit notes
~~~~~~~~~~~~~~~~~~~~~~

To send a debit or credit note to Infile, first create the :ref:`debit
<accounting/credit_notes/issue-debit-note>` or :ref:`credit note
<accounting/credit_notes/issue-credit-note>`. In the :guilabel:`Send` window, click
:guilabel:`Send to SAT (Guatemalan EDI)` to submit it for real-time validation. Upon successful validation,
the QR code from Infile is embedded in the debit or credit note PDF.
```

Export invoices
---------------

When creating exportation invoices, take into account the next considerations:

> - The customer's :guilabel:`Identification type` must be :guilabel:`VAT`, :guilabel:`Passport` or

Reviewer: Sure, to be on the same page: **_"Export Invoices"_** is a crucial document used in international trade workflows, this happens when a sales order is made for clients outside the country because it impacts the tax used and other relevant info. 

I know that the word can be confusing if you are thinking of an _'import/export_ data process' as Odooer, but in this case, for that section, we refer to a specific Invoicing document; it requires extra fields to be populated in Odoo, in order to be accepted/validated by the Government for electronic workflows.

That being said, this section, for Export Invoices, is another type of 'Electronic Invoicing Sending', not a sub part, or child of a 'Debit Note' type, as the structure is showing for now.

Let me know if that helped clarify the suggested change, thanks for your time and help =] @auva-odoo

=========
Guatemala
=========

.. |SAT| replace:: :abbr:`SAT (Superintendencia de Administración Tributaria)`
.. |EDI| replace:: :abbr:`EDI (Electronic Data Interchange)`

.. _guatemala/intro:

Introduction
============

With the Guatemalan localization, you can generate electronic documents with its XML, fiscal folio,

> electronic signature and connection to tax authority Superintendencia de Administración Tributaria

Reviewer 2: Nope, you're right, I based it on another loc, but we can change the wording, how about:

With the Guatemalan localization, you can connect to the tax authority Superintendencia de Administración Tributaria to generate electronic documents with its XML, fiscal folio, and electronic signature.

@auva-odoo @StraubCreative

(SAT) through Infile.

The supported documents are:

- :guilabel:`FACT-Factura`, :guilabel:`FCAM-Factura Cambiaria`, :guilabel:`NCRE-Credit Note`,
  :guilabel:`NDEB-Debit Note`;
- :guilabel:`NABN-Nota de Abono`, :guilabel:`FPEQ-Factura de Pequeño Contribuyente`,
  :guilabel:`FCAP-Factura Cambiaria Pequeño Contribuyente`;
- :guilabel:`FACT-Factura with Export Complement`.

> The localization requires an Infile account, which enables users to generate electronic documents

Author: @fvz-odoo

       above <guatemala/intro>`.

.. note::
   Odoo automatically installs the base module **Guatemala - Accounting** when a database is
   installed with `Guatemala` selected as the country. However, to enable electronic invoicing, the
   **Guatemala Accounting EDI** (`l10n_gt_edi`) module needs to be manually :ref:`installed
   <general/install>`.

Company
-------

To configure your company information, open the **Settings** app, scroll down to the
:guilabel:`Companies` section, click :guilabel:`Update Info`, and configure the following:

- :guilabel:`Company Name`
- :guilabel:`Address`, including the :guilabel:`Street`, :guilabel:`City`, :guilabel:`State`,
  :guilabel:`ZIP`, and :guilabel:`Country`
- :guilabel:`Tax ID`: Enter the identification number for the selected taxpayer type.
- :guilabel:`VAT Affiliation`: Select the VAT affiliation for the company, which is the type of
  Regime the company belongs to.
- :guilabel:`Legal Name`: The legal name of the company, which is used in the XML and PDF documents.
- :guilabel:`Establishment Code`: A necessary part of the XML when creating an electronic document.

>   If this field is not set, all electronic documents will be rejected.

Reviewer 2: I removed the part on the other lines, since this was the previous approach, now they can configure it directly in the Company Settings. @StraubCreative

> 

Reviewer 2: Add this, it was decided to also have it in res.company at the end.

```suggestion
- :guilabel: `Phrases`: The phrase  or phrases used in the XML document related to the VAT affiliation of your
  company. It is also used to generate the PDF document.
```

  To find the :guilabel:`Establishment Code`, follow these steps:

  #. From your `SAT account <https://portal.sat.gob.gt/>`_, go to :menuselection:`FEL -->
     Administración de Establecimientos`.
  #. You will see the list of establishments registered along with their corresponding code.

After configuring the company in the database settings, navigate to :menuselection:`Contacts` and
search for your company to verify the following:

> - the company type is set to :guilabel:`Company`.

Author: Revised as an admonition for the non-standard setting.
```suggestion
.. important::
   After configuring the company in the database settings, navigate to :menuselection:`Contacts` and
   verify that the :guilabel:`Identification Number` :guilabel:`Type` is set to :guilabel:`NIT`.
```

- the :guilabel:`Identification Number` :guilabel:`Type` is :guilabel:`NIT`.

> Go to the :guilabel:`Other Info` tab and verify that the following fields are set:
> 
> - :guilabel:`Phrase`: The phrase used in the XML document related to the VAT affiliation of your
>   company. It is also used to generate the PDF document.

Reviewer 2: ```suggestion
Go to the :guilabel:`Sales & Purchases` tab and verify under the :guilabel:`Fiscal Information` section that the following field is set:

- :guilabel:`Phrases`: The phrase or phrases used in the XML document related to the VAT affiliation of your
  company. It is also used to generate the PDF document.
```

Electronic Invoice Data
-----------------------

> 

Author: @fvz-odoo

> 

Author: Should be all set here: https://github.com/odoo/documentation/pull/14190#discussion_r2252498939

- Sign a service agreement directly with `Infile <https://infile.com.gt/>`_.
- Infile will provide you with the necessary credentials to input in Odoo.
- In Odoo, navigate to :menuselection:`Accounting --> Configuration --> Settings` and scroll down to
  the :guilabel:`Guatemalan Localization` section.
- Select the :guilabel:`Infile Web Services` environment, either :guilabel:`Production` or
  :guilabel:`Testing`.
- Enter the :guilabel:`Infile Data`:
  - :guilabel:`Infile WS Password`
  - :guilabel:`Commerce Code`
  - :guilabel:`Terminal Code`
- Click on :guilabel:`Save`.

.. note::
   The :guilabel:`Infile WS Password`, :guilabel:`Commerce Code`, and :guilabel:`Terminal Code` are
   provided by Infile. If you do not have them, contact Infile support.

.. tip::

>   - The demo environment is for testing only and does not generate legal documents, UUIDs, or fiscal

Reviewer 2: ```suggestion
  - The :guilabel:`Demo` environment is for testing only and does not generate legal documents, UUIDs, or fiscal
```

Please refer to the next section in our documentation for more information about
:doc:`multi-currencies <../accounting/get_started/multi_currency>`.

Master data
-----------

Chart of accounts
~~~~~~~~~~~~~~~~~

The :doc:`chart of accounts <../accounting/get_started/chart_of_accounts>` is installed by default
as part of the set of data included in the localization module, the accounts are mapped
automatically in taxes, default accounts payable, and default accounts receivable.

Accounts can be added or deleted according to the company's needs.

.. seealso::
   :doc:`../accounting/get_started/chart_of_accounts`

Contacts
~~~~~~~~

To create a contact, navigate to :menuselection:`Contacts app` and select :guilabel:`New`. Then
enter the following information:

- :guilabel:`Company Name`
- :guilabel:`Address`:

  - :guilabel:`Street`
  - :guilabel:`City`
  - :guilabel:`State`
  - :guilabel:`ZIP`
  - :guilabel:`Country`: required to confirm an electronic invoice.

- :guilabel:`Identification Number`:

  - :guilabel:`Type`: select a identification type.
  - :guilabel:`Number`: required to confirm an electronic invoice.

.. note::

>    If this contact will always need a specific :guilabel:`Phrase`, you can set it directly on the
>    contact form under the :guilabel:`Fiscal Information` section. This ensures that every electronic
>    invoice generated for this contact will automatically include the required phrase in the XML and
>    PDF documents.

Reviewer 2: ```suggestion
   If this contact will always need a specific :guilabel:`Phrase`, you can set it directly on the contact 
   form under the :guilabel:`Fiscal Information` section in the :guilabel:`Sales & Purchases` tab.
   This ensures that every electronic invoice generated for this contact will automatically include the 
   required phrase in the XML and PDF documents.
```

Taxes
~~~~~

As part of the Guatemala localization module, taxes are automatically created with its configuration
and related financial accounts.

.. image:: guatemala/taxes.png
   :alt: Taxes for Guatemala.

Workflows
=========

Once you have configured your database, you can create your electronic documents.

Sales documents
---------------

> Customer Electronic Invoices

Author: I think I did this as you asked, please take another look around line 201 when revisions are pushed up.

--------------------------

After the invoice confirmation, click :guilabel:`Send`. In the wizard that appears, make sure to
enable the :guilabel:`Send to SAT` and :guilabel:`Email` checkboxes to send the XML to the |SAT|
through Infile's web service and the validated invoice to the client email and click
:guilabel:`Send`. Then, the following occurs.

- The XML document is created.
- The UUID is generated.
- The XML is processed synchronously by Infile.

  - If accepted, the file is displayed in the chatter and the email to the client with the
    corresponding :file:`pdf` and :file:`xml` file is sent.
  - If rejected, the error message is displayed in a pop-up window showing the reason for the error
    and the email is not sent.

.. image:: guatemala/pdf-xml-chatter-guatemala.png
   :alt: EDI documents available in the chatter.

The :guilabel:`SAT` tab then displays the following:

- :guilabel:`Datetime`: Timestamp recorded of the XML creation.
- :guilabel:`GT Status`: Status result obtained in the |SAT| response. If the invoice was rejected,
  the error messages can be seen here.
- :guilabel:`UUID`: The unique identifier assigned by the |SAT| to the electronic document.

> - :guilabel:`Download`: To download the sent XML file, even if the |SAT| result was rejected.

Author: Should be fixed here: https://github.com/odoo/documentation/pull/14190#discussion_r2252519192

document that, when validated, is sent to |SAT| via Infile. It is necessary to have a validated
(posted) electronic invoice to register a credit note. On the invoice, click the :guilabel:`Credit
note` button to access the :guilabel:`Create credit note` form, then complete the following
information:

- :guilabel:`Reason`: Type the reason for the credit note.
- :guilabel:`Journal`: Select the journal.
- :guilabel:`Reversal Date`: Type the date.

Customer Electronic Debit Note
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The :doc:`Customer debit note <../accounting/customer_invoices/credit_notes>` is an electronic
document that, when validated, is sent to |SAT| via Infile. It is necessary to have a validated
(posted) electronic invoice to register a debit note. On the invoice, click the :icon:`fa-cog`
(:guilabel:`action menu`) icon, select the :guilabel:`Debit note` option to access the
:guilabel:`Create credit note` form, then complete the following information:

- :guilabel:`Reason`: Type the reason for the debit note.
- :guilabel:`Journal`: Select the journal.
- :guilabel:`Copy lines`: Tick the checkbox to copy the invoice lines to the debit note.

> - :guilabel:`Debit note date`: Type the date.

Author: Should be fixed here: https://github.com/odoo/documentation/pull/14190#discussion_r2252514364

.. note::
   Confirm the invoice to create it. To send the document to |SAT| via Infile, click on
   :guilabel:`Send` and select the checkbox :guilabel:`Send to SAT`.

Export invoices
***************

> When creating exportation invoices, take into account the next considerations:

Author: @fvz-odoo

> When creating exportation invoices, take into account the next considerations:

Reviewer 2: @auva-odoo Export Invoices are Invoices to foreign customers, they're different than local invoices and need special attention to some fields.

- The Identification type on your customer must be VAT, Passport or Foreign ID.
- Incoterms must be set on the customer invoice under the :guilabel:`Other Info` tab. Under the
  :guilabel:`Accounting` section.
- The Phrase type 4 code 1 must be set on the customer invoice under the :guilabel:`Other Info` tab,
  under the :guilabel:`Accounting` section.
- The taxes included in the invoice lines should be 0% taxes.
- The :guilabel:`Consignatory Company` field must be set on the customer invoice under the
  :guilabel:`Other Info` tab. Under the :guilabel:`Accounting` section.

.. image:: guatemala/guatemala-exp-invoice.png

>    :alt: Exportation invoices main data.

Reviewer: Here is a cropped version:
https://drive.google.com/file/d/1H4fhUdwwXi_ZJYPPSrPx70wmx79G0h0E/view?usp=drive_link

Ideally, if we can highlight the important fields, that will tend to be a common error/question, here is a second option of images selecting important fields:
https://drive.google.com/file/d/1TnvDQdDiZOc1gwxz_LNRvWJ26X8vmF9s/view?usp=drive_link

>    :alt: Exportation invoices main data.

Reviewer: Hi @auva-odoo 

I had listed both versions, ideally, the one with the boxes helps the user where to populate required fields in order to avoid errors
https://drive.google.com/file/d/1TnvDQdDiZOc1gwxz_LNRvWJ26X8vmF9s/view

<img width="1215" height="946" alt="image" src="https://github.com/user-attachments/assets/c971459a-42ab-415f-9434-d0e4ad9c4e8d" />
